packages/vite/src/node/plugins/define.ts

```diff
@@ -41,7 +41,7 @@ export function definePlugin(config: ResolvedConfig): Plugin {
     if (isBuild) {
       const match = key.match(metaEnvRe)
       if (match) {
-        userDefineEnv[match[1]] = `__vite__define__${userDefine[key]}`
+        userDefineEnv[match[1]] = `__vite__define__${key}__vite__define__`
```

Comment:
Maybe we could use something that isn't the same for the start/end of the placeholder? Maybe:
- `__vite__define__${key}__end__` or
- `__vite__define__${key}__define__vite__`

Reply:
LGTM. I prefer the latter.